javagen/src/main/java/com/azure/autorest/template/ProtocolSampleTemplate.java

```diff
@@ -0,0 +1,99 @@
+        // Import
+        List<String> imports = new ArrayList<>();
+        imports.add("com.azure.core.http.rest.PagedIterable");
+        imports.add("com.azure.core.http.rest.Response");
+        imports.add("com.azure.core.util.BinaryData");
+        imports.add("com.azure.identity.DefaultAzureCredentialBuilder");
+        javaFile.declareImport(imports);
+
+        int numParam = method.getParameters().size();
+        // Parameter values to pass in
+        List<String> params = new ArrayList<>();
+        for (int i = 0; i < numParam; i++) {
+            params.add("null");
+        }
+
+        StringBuilder binaryDataStmt = new StringBuilder();
+
+        List<String> requestOptionsStmts = new ArrayList<>();
+
+        example.getParameters().forEach((key, value) -> {
+            boolean match = false;
+            for (int i = 0; i < numParam; i++) {
+                ClientMethodParameter p = method.getParameters().get(i);
+                if (p.getName().equals(key)) {
+                    if (p.getClientType() != ClassType.BinaryData) {
+                        // Simple type
+                        params.set(i, '"' + value.getObjectValue().toString() + '"');
+                    } else {
+                        // BinaryData
+                        String binaryDataValue = '"' + value.getObjectToString().replace("\"", "\\\"") + '"';
+                        binaryDataStmt.append(String.format(
+                                "BinaryData %s = BinaryData.fromString(%s);", key, binaryDataValue));
+                        params.set(i, key);
+                    }
+                    match = true;
+                    break;
+                }
+            }
+            if (!match) {
+                method.getProxyMethod().getParameters().stream().filter(p -> p.getName().equals(key)).findFirst().ifPresent(p -> {
+                    if (p.getRequestParameterLocation() == RequestParameterLocation.Query) {
+                        requestOptionsStmts.add(String.format("requestOptions.addQueryParam(\"%s\", \"%s\");",
+                                key, value.getObjectValue().toString()));
+                    } else if (p.getRequestParameterLocation() == RequestParameterLocation.Header) {
+                        requestOptionsStmts.add(String.format("requestOptions.addHeader(\"%s\", \"%s\");",
+                                key, value.getObjectValue().toString()));
+                    }
+                });
+            }
+        });
+
+        String clientName = client.getInterfaceName() + "Client";
+        javaFile.publicClass(null, filename, classBlock -> {
+            classBlock.publicStaticMethod("void main(String[] args)", methodBlock -> {
+                String clientInit = "%s client = new %s()" +
+                        ".endpoint(System.getenv(\"ENDPOINT\"))" +
+                        ".credential(new DefaultAzureCredentialBuilder().build())" +
```

Comment:
This part might need some flexibility. Not all data-plane uses the `DefaultAzureCredential`. Some might be connection string; some might be credential key.

Reply:
How do we know what should be used? I notice there is a `securityDefinitions` in swagger. Most are implicit flows.

Reply:
This?
```
credential-types: tokencredential
credential-scopes: https://purview.azure.net/.default
```

Reply:
Done